101-hdinsight-rserver-ssh-password/azuredeploy.json

```diff
@@ -0,0 +1,180 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"location": {
+			"type": "string",
+			"allowedValues": ["Central US",
+			"East Asia",
+			"East US",
+            "East US 2",
+			"Japan East",
+			"Japan West",
+			"North Europe",
+			"South Central US",
+			"Southeast Asia",
+			"West Europe",
+			"West US"],
+			"metadata": {
+				"description": "The location where all azure resources will be deployed."
+			}
+		},
+		"clusterType": {
+			"type": "string",
+			"allowedValues": ["hadoop",
+			"spark"],
+			"metadata": {
+				"description": "The type of the HDInsight cluster to create."
+			}
+		},
+		"clusterName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the HDInsight cluster to create."
+			}
+		},
+		"clusterLoginUserName": {
+			"type": "string",
+			"metadata": {
+				"description": "These credentials can be used to submit jobs to the cluster and to log into cluster dashboards."
+			}
+		},
+		"clusterLoginPassword": {
+			"type": "securestring",
+			"metadata": {
+				"description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+			}
+		},
+		"sshUserName": {
+			"type": "string",
+			"metadata": {
+				"description": "These credentials can be used to remotely access the cluster."
+			}
+		},
+		"sshPassword": {
+			"type": "securestring",
+			"metadata": {
+				"description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+			}
+		},
+		"clusterStorageAccountName": {
+			"type": "string",
+			"metadata": {
+				"description": "The name of the storage account to be created and be used as the cluster's storage."
+			}
+		},
+		"clusterWorkerNodeCount": {
+			"type": "int",
+			"defaultValue": 4,
+			"metadata": {
+				"description": "The number of nodes in the HDInsight cluster."
+			}
+		}
+	},
+	"variables": {
+		"defaultApiVersion": "2015-05-01-preview",
+		"clusterApiVersion": "2015-03-01-preview"
+	},
+	"resources": [{
+		"name": "[concat(parameters('clusterName'),'/R-Server')]",
+		"type": "Microsoft.HDInsight/clusters/applications",
+		"apiVersion": "[variables('clusterApiVersion')]",
+		"dependsOn": ["[concat('Microsoft.HDInsight/clusters/',parameters('clusterName'))]"],
+		"properties": {
+			"marketPlaceIdentifier": "Microsoft.RServerForHDInsight.8.0.3",
+			"computeProfile": {
+				"roles": [{
+					"name": "edgenode",
+					"targetInstanceCount": 1,
+					"hardwareProfile": {
+						"vmSize": "Standard_D4_v2"
```

Comment:
Not for this version -- there will be a future improvement to HDInsight templates to have enums for node sizes.